src/gen_smtp_server_session.erl

```diff
@@ -908,8 +915,8 @@ receive_data(Acc, Transport, Socket, RecvSize, Size, MaxSize, Session, Options)
 		{error, timeout} ->
 			receive_data(Acc, Transport, Socket, 0, Size, MaxSize, Session, Options);
 		{error, Reason} ->
-			error_logger:error_msg("SMTP receive error: ~p~n", [Reason]),
-			exit(receive_error)
```

Comment:
those were propagated (via link) to a session process and the session process is "abruptly" killed:
```
[2020-10-29T08:00:52.750562+01:00 error <0.21202.231>] SMTP receive error: closed
[2020-10-29T08:00:52.751746+01:00 error <0.1007.0>] Ranch listener {gen_smtp_server,{0,0,0,0}} had connection process started with gen_smtp_server_session:start_link/4 at <0.19039.231> exit with reason: receive_error
```

Now this message is handled around line 230, callback module could decide what to do in the same way as for `setopts`. Session process will stop anyway by returning `{stop, ...}` now instead of just dying, which is good, because, eg, `terminate` gen_server callback will be called as well.